include/common.h

```diff
@@ -5,6 +5,16 @@
 
 #include <stdbool.h>
 
+// Identify the architecture by means of pre-defined macros
+#ifdef __aarch64__
```

Comment:
I'm not sure that redefining the includes like this is a good idea: it means that people now need to understand `ARCH` and its semantics. I'd suggest just using `#ifdef` in the normal code and getting rid of this define. As a happy bonus that will allow you to change the test above to be something like:

```c
#if defined(__aarch64__)
// do a dereference
#else if defined(__riscv)
// simply increase the bounds
#else
#error Platform not currently supported
#endif
```

In other words, you can change the test from dynamically guessing what should happen to statically enforcing it. That hopefully will also make it clearer to readers.